src/http/HttpRequestFactory.cpp

```diff
@@ -171,17 +171,23 @@ void MultipartBodyType(const std::string &msg, HttpRequest *request) {
 
     std::string bodyPart = msg.substr(startPos, endPos - startPos);
 
-    // Extract name and content from bodyPart
-    size_t namePos = bodyPart.find("name=\"") + 6;
-    size_t nameEndPos = bodyPart.find("\"", namePos);
-    std::string name = bodyPart.substr(namePos, nameEndPos - namePos);
-
-    size_t contentPos = bodyPart.find("\r\n\r\n") + 4;
-    std::string content =
-        bodyPart.substr(contentPos, bodyPart.size() - contentPos - 4);
-
-    bodyParts[name] = content;
-    startPos = endPos + boundary.length();
+    if (!bodyPart.empty()) {
```

Comment:
On cases like this, an 'early return' is a better approach

for example:

```c++
if (bodyPart.empty()) {
   return;
}
```
And the all of the logic necessary below, this way you don't need to indent the code another level and the reader can safely assume "if reach this piece of code than the request has a body"